controller/controller.js

```diff
@@ -2,74 +2,73 @@ const model = require('../models/user');
 const mongoose = require('mongoose');
 const bodyParser = require('body-parser');
 
-
-
 //  Add new data
 exports.adddata = async (req, res) => {
 
-    const uname = req.body.name;
-    const uemail = req.body.email;
-    console.log(".............data is ", req.body.name);
+    const { name: userName, email: userEmail, age: userAge, birthdate: userBdate, password: userPassword } = req.body;
     try {
-        // throw new Error('test error');
         await model.create(
             {
-                name: uname,
-                email: uemail
+                name: userName,
+                email: userEmail,
+                birthdate: userBdate,
+                age: userAge,
+                password: userPassword,
             }
         )
-        res.send("Data Added sucessfully").status(201);
+        res.status(201).send("Data Added sucessfully");
 
     } catch (err) {
         console.log(".................." + err.message);
-        res.send(err.message, 'Data Not Inserted').status(400);
+        // console.log(Object.keys(res))
+        // return new Error("sasdsa")
+        res.send(" Error in data Creation :" + err.message).status(400);
     }
 };
 
 // View the user data
 
-exports.viewuser = async (req,res) => {
-    console.log(req.body.name);
-    const username=req.body.name                                     ;
+exports.viewuser = async (req, res) => {
+    const username = req.params.username;
     try {
-        const data = await model.findOne({name:username}).lean();
-        console.log(data);
-
-        res.send("My Name is : " + data.name + "and My Email Addres is :" + data.email ).status(201);
+        const data = await model.findOne({ name: username }).lean();
+        // console.log(data);
+        res.send("Hear is My Details : \n   Name : " + data.name + "\n Email :" + data.email + "\n Age : " + data.age + "\n Birthdate : " + data.birthdate).status(201);
 
     } catch (err) {
-        res.send(err.message , 'Fetching data is unsucessfull').status(400);
+        res.send(err.message + 'Fetching data is unsucessfull').status(400);
     }
 }
 
-
 // Update the data using id 
 
 exports.updatedata = async (req, res) => {
-
-    const id = req.body.id;
-    const newname = req.body.name;
-    const newemail = req.body.email;
+    const username = req.params.username;
+    const { id, name: newname, email: newemail, age: newAge, birthdate: newBdate, password: newPassword } = req.body;
```

Comment:
same as above

Reply:
for update and delete types of endpoint we pass id from params not username, id should come from params not body